server/game/core/ability/abilityTargets/CardTargetResolver.ts

```diff
@@ -37,7 +37,7 @@ export class CardTargetResolver extends TargetResolver<ICardTargetResolver<Abili
               (properties.immediateEffect == null || properties.immediateEffect.hasLegalTarget(contextCopy) &&
                 (!properties.cardCondition || properties.cardCondition(card, contextCopy)));
         };
-        return CardSelector.for(Object.assign({}, properties, { cardCondition: cardCondition, targets: true }));
+        return CardSelectorFactory.for(Object.assign({}, properties, { cardCondition: cardCondition, targets: true }));
```

Comment:
nit: this is just a common programming convention, but if we're renaming this to "factory" then typically the corresponding method would be called "create" (it's kind of the de-facto method name used with the factory pattern)